src/utils/wallets/wallets.utils.ts

```diff
@@ -159,75 +161,103 @@ async function generateShareHash(share: string): Promise<string> {
   return Buffer.from(new Uint8Array(hashBuffer)).toString("base64");
 }
 
-export interface GenerateShareHashAndPublicKeyReturn {
+export interface GenerateShareHashAndEdKeysFromDeviceShareParams {
+  deviceShare: string;
+  session: DbSession;
+}
+
+export interface GenerateShareHashAndEdKeysFromRecoveryBackupShareParams {
+  recoveryBackupShare: string;
+  session: DbSession;
+}
+
+export type GenerateShareHashAndEdKeysParams =
```

Comment:
Used an union here so that we can tell apart what we are going to use this share for based on the param name (`deviceShare` vs `recoveryBackupShare`), as that affects the salt generation.